coda-prototype/initUI.js

```diff
@@ -445,21 +445,19 @@ function initUI(dataset) {
 
     $("#export-dataset").click(() => FileUtils.saveDataset(newDataset));
 
-    $("#dataset-file").on("change", event => { // Fires when the dataset file has been changed by the file picker UI
+    /**
+     * Handles loading a new dataset after the user has selected a new dataset file via the Dataset dropdown.
+     */
+    $("#dataset-file").on("change", event => {
         $(event.target).parents(".dropdown").removeClass("open");
 
-        // Hide the existing alert.
-        let alert = $("#alert");
-        alert[0].childNodes.forEach(node => {
-            if (node.nodeName === "#text") {
-                node.remove();
-            }
-        });
-        alert.hide();
-        $(".tableFloatingHeaderOriginal").show();
+        UIUtils.hideAlert(); // This is a safety just in case there is a code path below which doesn't display an alert.
 
         let files = $("#dataset-file")[0].files;
-        if (files.length > 0) {
+        if (files.length !== 1) {
+            UIUtils.displayAlertAsError("Too many files selected. Only one dataset file can be uploaded at once");
+            console.log("ERROR: Multiple files selected. Files were:", files);
+        } else {
```

Comment:
In general, I prefer a style of 'test the preconditions are ok, and if they aren't exit the function'. 

So by the time you're in the body of the function you can be confident it's ok. This also means one fewer levels of if-else nesting.

Reply:
changed.